src/lib/modes/aead/ccm/ccm.cpp

```diff
@@ -175,7 +175,7 @@ void CCM_Encryption::finish(secure_vector<byte>& buffer, size_t offset)
 
    for(size_t i = 0; i != ad.size(); i += BS)
       {
-      xor_buf(&T[0], &ad[i], BS);
+      xor_buf(T.data(), ad.data() + i, BS);
```

Comment:
I would revert this because i is meant to never reach ad.size() (see line 176). If it does, it is an error we'd like to catch during debugging.